datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -366,6 +432,17 @@ pub fn date_bin(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     }
 }
 
+enum Interval {
```

Comment:
Since we have this enum anyways, I wonder if there is any need for the trait?

For example, what if the code looked like

```rust
impl Inteval {
  // return time in nanoseconds that the source timestamp falls into based on the stride and origin
  fn bin(source: i64, origin: i64) -> i64 {
    match self {
      Self::Nanoseconds(v) => date_bin_nanos_interval(v, source, origin),
      Self::Months(v) => date_bin_months_interval(v, source, origin)
    }
  }
```

And then the  closure looks like
```
    let f = |x: Option<i64>| x.map(move |x| stride.bin( x, origin));
```

🤔 

I don't think you would need a trait at all

Reply:
It works. Thanks Andrew